src/extension.ts

```diff
@@ -39,16 +45,34 @@ class PSCompletionProvider implements CompletionItemProvider {
 }
 
 export async function activate(context: ExtensionContext) {
-  let registerCompletionProvider = languages.registerCompletionItemProvider(
-    'javascript',
-    new PSCompletionProvider(snippets as [Snippet]),
-    '*'
-  )
+  async function workspaceFoldersChange(folders: WorkspaceFolder[]) {
+    if (folders) {
+      const config = await prettier.resolveConfig(folders[0].uri.path)
+
+      // TODO: format snippets
+
+      if (disposable) {
+        disposable.dispose()
+      }
+
+      disposable = languages.registerCompletionItemProvider(
+        'javascript',
+        new PSCompletionProvider(snippets as [Snippet]),
+        '*'
+      )
+    }
+  }
+
+  let folders = workspace.workspaceFolders
+  await workspaceFoldersChange(folders)
+
+  // TODO: Somehow handle workspaces
+  // workspace.onDidChangeWorkspaceFolders(workspaceFoldersChange)
 }
 
 export function deactivate() {
```

Comment:
Why are we going with `deactivate` method instead of registering all disposable object with `context` ([link][1]).

[1]: https://code.visualstudio.com/docs/extensionAPI/vscode-api#_a-nameextensioncontextaspan-classcodeitem-id1139extensioncontextspan

Reply:
If we are going to use `onDidChangeWorkspaceFolders`, we need to cleanup after ourselves manually.

Reply:
I thought that's what `dispose` method is for. Aren't module scoped `deactivate` function and `context.subscriptions` triggered solely on extension deactivation, 'hence being same in effect?

Reply:
Yes but during the possible lifetime you can have multiple disposables.

Reply:
When you change a workspace folder, you can possibly have a different configuration and create a new `Snippets`. Why waiting for extension deactivation, when you can do that immediately? Of course this implementation makes sense after `TODO` below will be solved.

Reply:
Sure, but `deactivate` is at the end of it. Just like subscription scavenging.

Reply:
In the future `workspaceFoldersChange` can be possibly called multiple times. That's why it cleans old disposable before creating a new one (line 54).

Reply:
Just to clarify, by deactivation here we mean user disabling the extension or `code` quitting. Right?

Reply:
Yes.